ownCloud/Viewer/PDFViewerViewController.swift

```diff
@@ -20,44 +20,340 @@ import UIKit
+
+            constraints.append(containerView.leadingAnchor.constraint(equalTo: guide.leadingAnchor))
+            constraints.append(containerView.trailingAnchor.constraint(equalTo: thumbnailView.leadingAnchor))
+            constraints.append(containerView.bottomAnchor.constraint(equalTo: guide.bottomAnchor))
+
+        case .bottom:
+            constraints.append(thumbnailView.topAnchor.constraint(equalTo: containerView.bottomAnchor))
+            constraints.append(thumbnailView.leadingAnchor.constraint(equalTo: guide.leadingAnchor))
+            constraints.append(thumbnailView.trailingAnchor.constraint(equalTo: guide.trailingAnchor))
+            constraints.append(thumbnailView.bottomAnchor.constraint(equalTo: guide.bottomAnchor))
+            constraints.append(thumbnailView.heightAnchor.constraint(equalTo: self.view.heightAnchor, multiplier: thumbnailViewHeightMultiplier))
+
+            constraints.append(containerView.leadingAnchor.constraint(equalTo: guide.leadingAnchor))
+            constraints.append(containerView.trailingAnchor.constraint(equalTo: guide.trailingAnchor))
+            constraints.append(containerView.bottomAnchor.constraint(equalTo: thumbnailView.topAnchor))
+
+        case .none:
+            constraints.append(containerView.leadingAnchor.constraint(equalTo: guide.leadingAnchor))
+            constraints.append(containerView.trailingAnchor.constraint(equalTo: guide.trailingAnchor))
+            constraints.append(containerView.bottomAnchor.constraint(equalTo: guide.bottomAnchor))
+            thumbnailView.isHidden = true
+        }
+
+        self.activeViewConstraints = constraints
+
+    }
+
+    fileprivate func setupToolbar() {
+        searchButtonItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(search))
+        gotoButtonItem = UIBarButtonItem(image: UIImage(named: "ic_pdf_go_to_page"), style: .plain, target: self, action: #selector(goToPage))
+        outlineItem = UIBarButtonItem(image: UIImage(named: "ic_pdf_outline"), style: .plain, target: self, action: #selector(showOutline))
+
+        self.parent?.navigationItem.rightBarButtonItems = [
+            gotoButtonItem!,
+            searchButtonItem!,
+            outlineItem!]
+    }
+
+    fileprivate func selectPage(with label:String) {
+        guard let pdf = pdfView.document else { return }
+
+        if let pageNr = Int(label) {
+            if pageNr > 0 && pageNr < pdf.pageCount {
+                if let page = pdf.page(at: pageNr - 1) {
+                    self.pdfView.go(to: page)
+                }
+            }
+        }
+    }
+
+    fileprivate func updatePageLabel() {
+        guard let pdf = pdfView.document else { return }
+
+        guard let currentPageLabel = pdfView.currentPage?.label else { return }
+
+        pageCountLabel.text = "\(currentPageLabel) of \(pdf.pageCount)"
```

Comment:
@felix-schwarz good point!

Reply:
@felix-schwarz however.. I think there is a problem with your suggestion to use `PDFDocument.index(for: page)` since it returns 0-based index.. so if page numbers start with something like i, ii, iii, 1, this method would return value 3 for page labeled "1".

Reply:
@mneuwert Good point! The first page should be page 1 of course :) … since the `y` element of the "x of y" text is using the page count, the `x` should be the page offset (+1) instead of the label to stay consistent.

Reply:
@mneuwert I think for this "[page] of [total]" display at the bottom, labels are meaningless and prone to be inconsistent and confusing. Let's go for `PDFDocument.index(for: page) + 1`.